pallets/oracle-collection/src/lib.rs

```diff
@@ -217,43 +208,34 @@ pub mod pallet {
 				Error::<T>::IsNotAdmin
 			);
 
-			transactional::with_transaction(|| {
-				let result = Self::update_feeders(collection_id, key, feeders.clone());
-
-				// We do not want to apply the mutation,
-				// only check if there is no error in applying it
-				TransactionOutcome::Rollback(result)
-			})?;
-
-			T::ChangeGuard::note(collection_id, Change::Feeders(key, feeders).into())?;
+			T::ChangeGuard::note(collection_id, Change::CollectionInfo(info).into())?;
 
 			Ok(())
 		}
 
 		/// Apply an change previously proposed by
-		/// [`Pallet::propose_update_feeders`] if the conditions to get it ready
-		/// are fullfilled.
+		/// [`Pallet::propose_update_collection_info`] if the conditions to get
+		/// it ready are fullfilled.
 		///
 		/// This call is permissionless.
-		#[pallet::weight(T::WeightInfo::apply_update_feeders(T::MaxFeedersPerKey::get()))]
+		#[pallet::weight(T::WeightInfo::apply_update_collection_info(T::MaxFeedersPerKey::get()))]
 		#[pallet::call_index(1)]
-		pub fn apply_update_feeders(
+		pub fn apply_update_collection_info(
 			origin: OriginFor<T>,
 			collection_id: T::CollectionId,
 			change_id: T::Hash,
 		) -> DispatchResult {
 			ensure_signed(origin)?;
 
-			let Change::Feeders(key, feeders) = T::ChangeGuard::released(collection_id, change_id)?
+			let Change::CollectionInfo(info) = T::ChangeGuard::released(collection_id, change_id)?
 				.try_into()
 				.map_err(|_| Error::<T>::NoOracleCollectionChangeId)?;
 
-			Self::update_feeders(collection_id, key, feeders.clone())?;
+			CollectionInfo::<T>::insert(collection_id, info.clone());
```

Comment:
I think we should forbid any non-allowed feeder to feed any further values at this point. Basically, iterate over all keys in the storage and remove unknown feeders.

Reply:
This is a very good point

> Basically, iterate over all keys in the storage and remove unknown feeders.

I would say this could be more complex because some existing keys could have lost some of their feeders (but not all), and then the aggregated values could have changed or not been validated.

Could we consider that if we call `apply_update_collection_info()`, then you should call later `update_collection()` to reflect those changes? We can understand the current collection as the values the collection had at the moment of updating. If we see this in such a way, then there is no problem if feeders are modified because the collection at that point contained such previous feeders and values.

Reply:
If you agree with me, it could be interesting to add a `last_updated` field to the collection to know if the content of the collection reflects a moment in the past.

Reply:
> to add a last_updated field to the collection to know if the content of the collection reflects a moment in the past.

I thought that is already the case?

Reply:
>Could we consider that if we call apply_update_collection_info(), then you should call later update_collection() to reflect those changes? We can understand the current collection as the values the collection had at the moment of updating. If we see this in such a way, then there is no problem if feeders are modified because the collection at that point contained such previous feeders and values.

Yes, that is a good idea. `update_collection` will filter them already, right?

Reply:
Maybe, `apply_update_collection_info` should just purge the `CachedCollection` storage?

Reply:
I think it makes more sense to purge it 👍🏻

Reply:
> I thought that is already the case?

The timestamp stored in the collection is the timestamp of the older Oracle value, not the timestamp of when the collection was outdated. So now we will store both

Reply:
So purge or store outdated timestamp?

Reply:
I think we can use both. Store the `last_timestamp` each time we call `update_collection()` and purge when apply.